Code-4-Community/speak-for-the-trees-frontend
# Sd.mobile landing page

## Why

[Monday.com Ticket](https://code4community-team.monday.com/boards/925293934/pulses/925485550)


Allows the user to navigate the landing page of the site on a mobile device more easily. Creates components that can be reused when creating other mobile pages.

## This PR


Global Styling
- renamed `DARKGREY` to `DARK_GREY` for consistency
- added `BACKGROUND_GREY`

Footer for Mobile
- created `MobileFooter` to match footer in mobile designs

Nav Bar
- created a separate component for `MobileNavBar` (separate component to improve readability)
- `NavBar` determines if the device is mobile and whether the user is logged in, renders `MobileNavBar` if mobile and passes `isLoggedIn`
- mobile nav bar has a dropdown menu if user is logged in

Create map page for mobile
- created `MobileMapPage` component
- compared to `MapPage`, children are displayed below the map rather than as a sidebar
- scroll down to see `MobileFooter`

Optimize map view for mobile
- `MapView` does not render address search bar if mobile

Create landing bar for mobile
- created `MobileLandingBar` component to match designs

Optimize landing page for mobile
- renders `MobileMapPage` if mobile

Create info card for mobile
- created `MobileInfoCard` component to match landing stats card in mobile designs (rather than optimizing `InfoCard` to reduce use of `useWindowDimensions`)

Optimize landing stats for mobile, use styled components
- deleted `landing-tree-stats.less` and used styled components
- displays stats in `MobileInfoCard` if mobile, otherwise in `InfoCard`

## Screenshots

 
Mobile Landing Page with Mobile Nav Bar (left), Mobile Landing Page with Mobile Footer (right)
<img width="200" alt="MobileLanding" src="https://user-images.githubusercontent.com/22990100/106402315-efba4a00-63f6-11eb-8f73-a4e8b6c9db11.png">  <img width="200" alt="MobileFooter" src="https://user-images.githubusercontent.com/22990100/106402317-f1840d80-63f6-11eb-9932-ef93cd55c2cb.png">
Dropdown Menu on Mobile Nav Bar (displayed when logged in)
<img width="375" alt="MobileNavDropdown" src="https://user-images.githubusercontent.com/22990100/106402322-f3e66780-63f6-11eb-839c-93df9236bf88.png">

## Verification Steps

 

Used Device Mode in Chrome Dev Tools to check the mobile layout and map functionality, clicked the dropdown menu and links to make sure they worked.
